src/runtimes/docker/utils/spawnStreamAsync.ts

```diff
@@ -229,7 +231,41 @@ export async function spawnStreamAsync(
     }
 
     if (options.onCommand) {
-        options.onCommand([command, ...args].join(' '));
+        options.onCommand([command, ...normalizedArgs].join(' '));
+    }
+
+    let stdOutPipe: NodeJS.WritableStream | undefined = options.stdOutPipe;
+    let stdErrPipe: NodeJS.WritableStream | undefined = options.stdErrPipe;
+    if (ext.diagnosticLogging) {
+        ext.outputChannel.appendLine([command, ...normalizedArgs].join(' '));
+
+        const diagnosticStdOutPipe = new stream.PassThrough();
+        diagnosticStdOutPipe.on('data', (chunk: Buffer) => {
+            try {
+                ext.outputChannel.appendLine(chunk.toString());
+            } catch {
+                // Do not throw on diagnostic errors
+            }
+        });
+
+        if (stdOutPipe) {
+            diagnosticStdOutPipe.pipe(stdOutPipe);
+        }
+        stdOutPipe = diagnosticStdOutPipe;
+
+        const diagnosticStdErrPipe = new stream.PassThrough();
+        diagnosticStdErrPipe.on('data', (chunk: Buffer) => {
+            try {
+                ext.outputChannel.appendLine('stderr: ' + chunk.toString());
+            } catch {
+                // Do not throw on diagnostic errors
+            }
+        });
+
+        if (stdErrPipe) {
+            diagnosticStdErrPipe.pipe(stdErrPipe);
+        }
+        stdErrPipe = diagnosticStdErrPipe;
     }
 
     const childProcess = spawn(
```

Comment:
I think we should dump the entire environment as well, i.e. `{...process.env, ...options.env}`.

Reply:
At a minimum, any `DOCKER_*` environment variables and `PATH`.

Reply:
Logging the full environment (process and configured) makes sense; I’ll update what we output here. While theoretically a user could change the docker environment setting mid run, I’m leaning towards logging once at activation as a reasonable trade off between fidelity and noise in diagnostic logging.

Reply:
Updated to log environment at debug level.